### PR TITLE
When search expands, auto focus on the input field.

### LINKED ISSUE #373: autofocus on search input when it is shown

### PR DESCRIPTION
**Pull request by bjedrzejewski**
Closes #373

### CODE REVIEW

**Comment by akhmorgan** [comment]
looks good to me.

File: src/sidebars/sidebar-controller.js [hunk 1]
```diff
@@ -51,8 +54,25 @@
                 this._showFavourites = true;
             }
         }
+
+        _showSearchesChanged() {
+            var showSearches = this.showSearches();
+            if (showSearches) {
+                //Couldn't get it to work reliable in a synchronous/promise fashion
+                setTimeout(function() {
```
Comment (owennw) [inline src/sidebars/sidebar-controller.js]: `setTimeout` shouldn't be used in angular apps. Instead inject `$timeout` and use that. Also would prefer the ES6 style function syntax.

File: src/sidebars/sidebar-controller.js [hunk 2]
```diff
@@ -51,8 +54,25 @@
                 this._showFavourites = true;
             }
         }
+
+        _showSearchesChanged() {
+            var showSearches = this.showSearches();
+            if (showSearches) {
+                //Couldn't get it to work reliable in a synchronous/promise fashion
```
Comment (owennw) [inline src/sidebars/sidebar-controller.js]: Not sure if you need the apologetic comment :P

Comment (bjedrzejewski) [inline src/sidebars/sidebar-controller.js]: This is a disguised 'todo' for someone other than myself :D

**Comment by owennw** [comment]
Seems like the feature doesn't work in compact mode, unfortunately. But it's good in normal!

**Comment by akhmorgan** [comment]
It won't work in compact view because it's getting the input by id and the compact view has it's own search input, which is the second element in the sidebar html with that id. So it's only ever getting the full-view search input.

**Comment by WPFerg** [comment]
Perhaps having something in the directive on expand would be more straight forward, then?

**Comment by bjedrzejewski** [comment]
@akhmorgan where is that second input in the code? Which file?

**Comment by akhmorgan** [comment]
@bjedrzejewski in sidebar.html you'll see two `<search>` elements

**Comment by bjedrzejewski** [comment]
@akhmorgan ok I got it now. I have a nice fix I think.

**Comment by bjedrzejewski** [comment]
@WPFerg @akhmorgan I will move to react for a bit now. If anyone want's to fix it (I failed without jQuery for now), feel free to add a commit!

**Comment by WPFerg** [comment]
I think I've got a cleaner fix doing something different, I'll close this and open a new PR with it.

**Pull request closed without merging**